py/picca/delta_extraction/astronomical_objects/desi_pk1d_forest.py

```diff
@@ -197,22 +197,33 @@ def rebin(self):
         AstronomicalObjectError if Forest.wave_solution is not 'lin' or 'log'
         """
         bins, rebin_ivar, orig_ivar, w1, w2 = super().rebin()
-        if len(rebin_ivar) == 0:
+        if len(rebin_ivar) == 0 or np.sum(w2) == 0:
             self.resolution_matrix = np.array([[]])
-            return [], [], [], [], []
+            return [], [], [], [], np.array([])
 
         # apply mask due to cuts in bin
         self.resolution_matrix = self.resolution_matrix[:, w1]
 
+        # Find non-empty bins
+        binned_arr_size = bins.max() + 1
+        bincounts = np.bincount(bins, minlength=binned_arr_size)
+        wnonempty_bins = bincounts != 0
+
+        # Do a simple average when ivar=0
+        orig_ivar_2 = orig_ivar[w1]
+        w__ = orig_ivar_2>0
+        orig_ivar_2[~w__] = 1
+        rebin_reso_ivar = np.bincount(bins, weights=orig_ivar_2, minlength=binned_arr_size)
```

Comment:
so this gives masked pixels a weight of 1 in the rebinned reso matrix whereas for the non-masked pixels ivar weighting is used, leading to potentially lower weight than for a masked pixel (i.e. whenever ivar<1)?

Reply:
I guess the whole resolution matrix rebinning is anyway nonsense when actually rebinning pixels, as something would need to happen in the other dimension (i.e. the position difference wrt the central pixel) of the resolution matrix as well, so it's not really an issue, but just wanted to point this out
I guess in principle we'd also want to somehow flag the resolution matrix elements that are affected by a masked pixel in any way, i.e. in both the central pixel dimension or the difference dimension, but that might not be worth the effort

Reply:
Interesting point. The resolution matrix values are still needed even when `ivar=0`, so this was my solution to keep them. We can go back to this later

Reply:
Agreed, for the moment the resolution matrix values are still needed for central pixel with `ivar!=0` and offset pixel `ivar==0`, not sure if there's a way around this anyway... Regarding central pixels with `ivar==0` I guess we don't need anything, right? 

And regarding the weighting at rebinning stage I'd probably ignore masked pixels, but as said when we're actually rebinning the concept of using the resolution matrix doesn't really make sense (without at least a "rebinning" of the resolution matrix in both its dimensions, that is, which would then produce the kind of issues we saw in the quickquasars reso matrix).